### PR TITLE
Fix pagination

### LINKED ISSUE #288: Pagination doesn't remembers search terms
When you search for a user and click the pagination when you have quite a few results, the search term will be lost.

This also happens when searching for threads in the forum.

### PR DESCRIPTION
**Pull request by LucasLeandro1204**
Fix #288

### CODE REVIEW

**Comment by LucasLeandro1204** [comment]
Maybe is better to append inside the Search class? @driesvints

**Comment by driesvints** [comment]
@LucasLeandro1204 yes I think so too but I would only append the search result. That way you don't have to pass the request to the query object which could be a little weird.

File: app/Http/Controllers/Admin/AdminController.php [hunk 1]
```diff
@@ -18,6 +18,7 @@ public function __construct()
     public function index()
     {
         $users = request()->has('search') ? SearchUsers::get(request('search')) : User::findAllPaginated();
+        $users->appends(request()->except('page'));
```
Comment (driesvints) [inline app/Http/Controllers/Admin/AdminController.php]: Let's indeed append this inside the `SearchUsers` object. Just append the search result, not the entire request. Do this for the `SearchThreads` as well.

**Comment by LucasLeandro1204** [comment]
@driesvints done

File: app/Queries/SearchThreads.php [hunk 2]
```diff
@@ -15,6 +15,7 @@ public static function get(string $keyword, int $perPage = 20): Paginator
         return Thread::feedQuery()
             ->where('threads.subject', 'LIKE', "%$keyword%")
             ->orWhere('threads.body', 'LIKE', "%$keyword%")
-            ->paginate($perPage);
+            ->paginate($perPage)
+            ->appends(request()->only('search'));
```
Comment (driesvints) [inline app/Queries/SearchThreads.php]: Don't use the request inside a query object. Use the keyword like so:

```
->appends(['search' => $keyword);
```

File: app/Queries/SearchUsers.php [hunk 3]
```diff
@@ -12,10 +12,10 @@ class SearchUsers
...
-        return User::where(function ($query) use ($keyword) {
-            $query->where('name', 'like', "%$keyword%")
-                ->orWhere('email', 'like', "%$keyword%")
-                ->orWhere('username', 'like', "%$keyword%");
-        })->paginate($perPage);
+        return User::where('name', 'like', "%$keyword%")
+            ->orWhere('email', 'like', "%$keyword%")
+            ->orWhere('username', 'like', "%$keyword%")
+            ->paginate($perPage)
+            ->appends(request()->only('search'));
```
Comment (driesvints) [inline app/Queries/SearchUsers.php]: Don't use the request inside a query object. Use the keyword like so:

```
->appends(['search' => $keyword);
```

Comment (LucasLeandro1204) [inline app/Queries/SearchUsers.php]: Ah, ok =)

**Comment by driesvints** [comment]
Perfect, thanks! :)

**Pull request merged**